mart/attack/perturber/perturber.py

```diff
@@ -56,6 +59,13 @@ def projector_wrapper(perturber_module, args):
         if projector is not None:
             self.register_forward_pre_hook(projector_wrapper)
 
+    def parameters_optim(self):
+        """Return parameters along with the pre-defined optimization parameters.
+
+        Example: `{"params": perturbation, "lr":0.1, "momentum": 0.9}`
+        """
+        return [{"params": self.perturbation} | self.optim_params]
```

Comment:
You might want to make sure that `self.optim_params` does not contain a `params` key.

Reply:
Added.